tests/integration_test.py

```diff
@@ -67,30 +74,275 @@ def test_attachment_plugin_new_record(
+
+    record = client.get_record(id="logo", bucket="blog", collection="articles")
+
+    assert record
+    assert "data" in record
+    assert "attachment" in record["data"]
+
+
+def test_signer_plugin(
+    get_clients: Tuple[Client, Client, Client],
+    server: str,
+    source_bucket: str,
+    source_collection: str,
+    reset: bool,
+):
+    client, editor_client, reviewer_client = get_clients
+
+    # 0. initialize source bucket/collection (if necessary)
+    server_info = client.server_info()
+    editor_id = editor_client.server_info()["user"]["id"]
+    reviewer_id = reviewer_client.server_info()["user"]["id"]
+    print("Server: {0}".format(server))
+    print("Author: {user[id]}".format(**server_info))
+    print("Editor: {0}".format(editor_id))
+    print("Reviewer: {0}".format(reviewer_id))
+
+    # 0. check that this collection is well configured.
+    signer_capabilities = server_info["capabilities"]["signer"]
+
+    resources = [
+        r
+        for r in signer_capabilities["resources"]
+        if (source_bucket, source_collection)
+        == (r["source"]["bucket"], r["source"]["collection"])
+        or (source_bucket, None) == (r["source"]["bucket"], r["source"]["collection"])
+    ]
+    assert resources, "Specified source not configured to be signed"
+    resource = resources[0]
+    msg = (
+        "Signoff: {source[bucket]}/{source[collection]} => "
+        "{preview[bucket]}/{preview[collection]} => "
+        "{destination[bucket]}/{destination[collection]} "
+    )
+    print(msg.format(**resource))
+    print("_" * 80)
+
+    client.create_bucket(if_not_exists=True)
+    client.create_bucket(id=resource["preview"]["bucket"], if_not_exists=True)
+    bucket = client.create_bucket(
+        id=resource["destination"]["bucket"], if_not_exists=True
+    )
+
+    client.create_collection(
+        permissions={
+            "write": [editor_id, reviewer_id] + bucket["permissions"]["write"]
+        },
+        if_not_exists=True,
+    )
+
+    editors_group = (
+        resource.get("editors_group") or signer_capabilities["editors_group"]
+    )
+    editors_group = editors_group.format(collection_id=source_collection)
+    client.patch_group(id=editors_group, data={"members": [editor_id]})
+
+    reviewers_group = (
+        resource.get("reviewers_group") or signer_capabilities["reviewers_group"]
+    )
+    reviewers_group = reviewers_group.format(collection_id=source_collection)
+    client.patch_group(id=reviewers_group, data={"members": [reviewer_id]})
+
+    if reset:
+        client.delete_records()
+        existing = 0
+    else:
+        existing_records = client.get_records()
+        existing = len(existing_records)
+
+    dest_col = resource["destination"].get("collection") or source_collection
+    dest_client = Client(
+        server_url=server,
+        bucket=resource["destination"]["bucket"],
+        collection=dest_col,
+    )
+
+    preview_client = None
+    if "preview" in resource:
```

Comment:
Maybe we can remove this (see #1946). We could assume that we are testing the usual Remote Settings setup `main-workspace/{cid} -> main-preview/{cid} -> main/{cid}`

Reply:
https://github.com/mozilla-services/kinto-dist/pull/1953/commits/ae50147473252fcc96d92271340469a55e9e8109